coramin/third_party/minlplib_tools.py

```diff
@@ -0,0 +1,336 @@
+import os
+from zipfile import ZipFile
+from pyomo.common import fileutils
+from pyomo.common import download
+import enum
+import math
+import csv
+from collections.abc import Iterable
+import logging
+
+
+logger = logging.getLogger(__name__)
+
+
+def get_minlplib_instancedata(download_dir=None):
+    """
+    Downlaod instancedata.csv from MINLPLib which can be used to get statistics on the problems from minlplib.
+
+    Parameters
+    ----------
+    download_dir: str
+        The directory in which to place the downloaded file. The default will be a 
+        directory called minlplib in the current working directory.
+    """
+    if download_dir is None:
+        download_dir = os.path.join(os.getcwd(), 'minlplib')
+
+    if not os.path.exists(download_dir):
+        os.makedirs(download_dir)
+
+    downloader = download.FileDownloader()
+    downloader.set_destination_filename(os.path.join(download_dir, 'instancedata.csv'))
+    downloader.get_binary_file('http://www.minlplib.org/instancedata.csv')
```

Comment:
This is weird: you are using a method designed for binary files to fetch a text file.  I will put in a patch to Pyomo so that there is a `get_text_file` method.

Reply:
I tried `get_text_file`, but I got an exception about expecting `str` not `bytes`. Any thoughts?

Reply:
Can you give me an example (email is fine)?  I ran into this with the tests for `get_text_file` and thought that I resolved it...

Reply:
fixed

Reply:
I thought this was fixed, but I had to switch back to `get_binary_file`. Is there a problem with using `get_binary_file`?